# Tarefa 1

Seja uma rede 802.11b instalada em um escritório.

Considere pouca mobilidade (pessoas caminhando a 1 m/s), portadora de 2.4 GHz e banda de 20MHz.

Assuma um espalhamento de atraso rms de 0.07µs e que uma mensagem (um frame com vários símbolos) dura menos que 19ms.

- Determine o tempo e a banda de coerência.
- Classifique o canal em termos de seletividade em frequência e variação temporal.


In [1]:
import numpy as np

# Dados fornecidos
v = 1  # velocidade em m/s
fc = 2.4e9  # frequência da portadora em Hz
c = 3e8  # velocidade da luz em m/s
tau_rms = 0.07e-6  # espalhamento de atraso rms em segundos

In [2]:
# Calcular o comprimento de onda
lambda_ = c / fc

# Calcular o desvio Doppler máximo
fd = (v * fc) / c

# Calcular o tempo de coerência
Tc = lambda_ / (v)

# Calcular a largura de banda de coerência
Bc = 1 / (5 * tau_rms)

In [3]:
print(f"Tempo de coerência: {Tc*1000:0.0f} ms")

print(f"Comprimento de onda: {lambda_:.2f} m")


print(f"Desvio Doppler máximo: {fd:.2f} Hz")

print(f"Largura de banda de coerência: {Bc/1e6:.3f} MHz")

Tempo de coerência: 125 ms
Comprimento de onda: 0.12 m
Desvio Doppler máximo: 8.00 Hz
Largura de banda de coerência: 2.857 MHz


## Cálculo dos Parâmetros do Canal

Comprimento de Onda

$$
\lambda = \frac{c}{f_c} = \frac{3 \times 10^8 \, \text{m/s}}{2{,}4 \times 10^9 \, \text{Hz}} = 0{,}125 \, \text{m}
$$

Desvio Doppler Máximo

$$
f_d = \frac{v \cdot f_c}{c} = \frac{1 \, \text{m/s} \times 2{,}4 \times 10^9 \, \text{Hz}}{3 \times 10^8 \, \text{m/s}} = 8 \, \text{Hz}
$$

Tempo de Coerência

$$
T_c = \frac{\lambda}{v} = \frac{0{,}125 \, \text{m}}{1 \, \text{m/s}} = 0{,}125 \, \text{s} = 125 \, \text{ms}
$$

Largura de Banda de Coerência

$$
B_c = \frac{1}{5 \cdot \tau_{\text{rms}}} = \frac{1}{5 \times 0{,}07 \times 10^{-6} \, \text{s}} = 2{,}857 \, \text{MHz}
$$

## Classificação do Canal

Observando os resultados, pode-se concluir que o **canal é lento**, pois o tempo de coerência é de **125 ms** e é bem **maior** que o tempo de duração da mensagem, que é de **19 ms**. No entanto, este canal é **seletivo em frequência**, pois a banda de coerência (2,857 MHz) é menor que a banda do sinal (20 MHz).


# Tarefa 2

Seja o seguinte perfil de potências de um canal de TV digital:

| Multipercurso      | 1   | 2   | 3   | 4   | 5   | 6    |
| ------------------ | --- | --- | --- | --- | --- | ---- |
| **Atraso (µs)**    | 0   | 0.3 | 3.5 | 4.4 | 9.5 | 12.7 |
| **Atenuação (dB)** | 0   | 12  | 4   | 7   | 15  | 22   |

1. Determine a banda de coerência.
2. Sabendo que o sinal de TV digital tem banda de aproximadamente 6MHz, o canal é seletivo em frequência?


In [4]:
# Dados fornecidos
atrasos = np.array([0, 0.3, 3.5, 4.4, 9.5, 12.7])  # em microsegundos
atenuacoes_db = np.array([0, 12, 4, 7, 15, 22])  # em dB

In [5]:
# Passo 1: Converter atenuações de dB para potência linear
# Fórmula: P_linear = 10^(-A_dB / 10)
potencias = 10 ** (-atenuacoes_db / 10)

# Passo 2: Calcular o atraso médio (tau_mean)
tau_mean = np.sum(potencias * atrasos) / np.sum(potencias)

# Passo 3: Calcular o atraso médio quadrado (tau_mean^2)
tau_mean_squared = np.sum(potencias * atrasos**2) / np.sum(potencias)

# Exibir os resultados
print(f"Atraso médio (tau_mean): {tau_mean:.4f} microsegundos")
print(
    f"Atraso médio quadrado (tau_mean_squared): {tau_mean_squared:.4f} microsegundos\u00b2"
)

Atraso médio (tau_mean): 1.5723 microsegundos
Atraso médio quadrado (tau_mean_squared): 7.4276 microsegundos²


In [21]:
sigma_rms = np.sqrt(tau_mean_squared - tau_mean**2)
print(f"Espalhamento de atraso rms: {sigma_rms:.4f} microsegundos")

Espalhamento de atraso rms: 2.2261 microsegundos


In [24]:
# Passo 4: Calcular a banda de coerência (Bc)
# Converter tau_rms de microsegundos para segundos antes de calcular
sigma_rms_s = sigma_rms * 1e-3
Bc_2 = 1 / (5 * sigma_rms_s)

print(f"Banda de coerência (Bc): {Bc_2:.4f} kHz")

Banda de coerência (Bc): 89.8432 kHz


Como a Banda de coerencia é menor que oa largura de banda do canal, logo o canal é seletivo em frequência.


## Cálculo dos Parâmetros do Canal

Atraso Médio (τ_mean)

$$
\tau_{\text{mean}} = \frac{\sum (P_{\text{linear}} \times \text{atrasos})}{\sum P_{\text{linear}}} = 1{,}5723 \, \text{µs}
$$

Atraso Médio Quadrado ($\tau^2_{mean}$)

$$
\tau_{\text{mean}}^2 = \frac{\sum (P_{\text{linear}} \times \text{atrasos}^2)}{\sum P_{\text{linear}}} = 7{,}4276 \, \text{µs}^2
$$

Espalhamento de Atraso RMS ($\sigma_{T}$)

$$
\sigma_{T} = \sqrt{\tau_{\text{mean}}^2 - \tau_{\text{mean}}^2} = 2{,}2261 \, \text{µs}
$$

Banda de Coerência (Bc)

$$
B_c = \frac{1}{5 \cdot \sigma_{T}} = \frac{1}{5 \times 2{,}2261 \times 10^{-6} \, \text{s}} = 89{,}84 \, \text{kHz}
$$

## Classificação do Canal

Com base nos cálculos, podemos concluir que este **canal é seletivo em frequência**, pois a banda de coerência (89,84 kHz) é **menor** que a largura de banda do sinal (6 MHz).
